test/cpp/tensorexpr/test_cuda.cpp

```diff
@@ -161,8 +161,8 @@ void testCudaDynamicShape2D() {
   auto testWithSize = [](int32_t M, int32_t N) {
     VarHandle m("m", kInt);
     VarHandle n("n", kInt);
-    Buffer a(VarHandle("a", kHandle), kFloat, {m, n});
-    Buffer b(VarHandle("b", kHandle), kFloat, {m, n});
+    Buffer a(BufHandle("a", kHandle, {m, n}), kFloat);
```

Comment:
BufHandle takes a kHandle is a bit weird. It is somewhat expected for the original VarHandle, since it is a much lower level API. Some unfamiliar might expect BufHandle would imply its own handle value, and the Dtype is that of the value, or the handle itself.

Reply:
Right, I noticed that part too and it seemed redundant, but I wasn't sure it was. I can remove the `kHandle` argument I think.